### PR TITLE
Handle Nullpointer when schema name is null

### LINKED ISSUE #18039: NullPointerException on querying system.jdbc.columns
Presto version: 0.273.4
postgresql: 9.6.11

When attempting to run the query `SELECT * FROM system.jdbc.columns`, I am getting a `NullPointerException`:

```
2022-07-15T02:04:38.568-0400	ERROR	remote-task-callback-13	com.facebook.presto.execution.StageExecutionStateMachineStage execution 20220715_060438_00006_gjk98.1.0 failed
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at com.facebook.presto.plugin.jdbc.JdbcMetadata.listTableColumns(JdbcMetadata.java:137)
	at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorMetadata.listTableColumns(ClassLoaderSafeConnectorMetadata.java:308)
	at com.facebook.presto.metadata.MetadataManager.listTableColumns(MetadataManager.java:595)
	at com.facebook.presto.metadata.MetadataListing.listTableColumns(MetadataListing.java:95)
	at com.facebook.presto.connector.system.jdbc.ColumnJdbcTable.cursor(ColumnJdbcTable.java:126)
	at com.facebook.presto.connector.system.SystemPageSourceProvider$1.cursor(SystemPageSourceProvider.java:130)
	at com.facebook.presto.split.MappedRecordSet.cursor(MappedRecordSet.java:53)
	at com.facebook.presto.spi.RecordPageSource.<init>(RecordPageSource.java:40)
	at com.facebook.presto.connector.system.SystemPageSourceProvider.createPageSource(SystemPageSourceProvider.java:109)
	at com.facebook.presto.spi.connector.ConnectorPageSourceProvider.createPageSource(ConnectorPageSourceProvider.java:52)
	at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:80)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:259)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:424)
	at com.facebook.presto.operator.Driver.lambda$processFor$9(Driver.java:307)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:728)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:300)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1079)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:166)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:599)
	at com.facebook.presto.$gen.Presto_0_273_4_3907bd6____20220715_055745_1.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

I can run other queries against the database just fine, including `SELECT * FROM information_schema.columns` and querying against the actual data in the DB, but presto just doesn't like something about `system.jdbc.columns`.

### PR DESCRIPTION
**Pull request by agrawalreetika**
Fixes #18039

### CODE REVIEW

**Comment by jainxrohit** [comment]
Can you please look in to the build failures?

**Comment by agrawalreetika** [comment]
> 

Hi @jainxrohit, All checks are green, please have a look.

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcMetadata.java [hunk 1]
```diff
@@ -130,12 +130,15 @@ public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSess
     {
         ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
         List<SchemaTableName> tables;
-        if (prefix.getTableName() != null) {
+        if (prefix.getSchemaName() != null && prefix.getTableName() != null) {
             tables = ImmutableList.of(new SchemaTableName(prefix.getSchemaName(), prefix.getTableName()));
         }
-        else {
+        else if (prefix.getSchemaName() != null) {
```
Comment (ajaygeorge) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcMetadata.java:136]: We can use Optional.ofNullable here

**Review by ajaygeorge — APPROVED** [review]
LGTM % a comment.
Did we find out why the schema is coming as null in the postgres use case?

**Comment by agrawalreetika** [comment]
> LGTM % a comment. Did we find out why the schema is coming as null in the Postgres use case?

I haven't dug deeper, but I noticed in Metadata calls for listing columns schema and tables names are empty for any catalog when the system catalog is being queried.

**Comment by agrawalreetika** [comment]
@ajaygeorge Is it good to merge?

**Comment by ajaygeorge** [comment]
@agrawalreetika yup. LGTM. Already approved.

**Comment by agrawalreetika** [comment]
@ajaygeorge Can you merge this? Or someone else has to do it?

**Comment by ajaygeorge** [comment]
> @ajaygeorge Can you merge this? Or someone else has to do it?

Adding committer, @tdcmeehan can you please take a look.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**